tools/runner

```diff
@@ -116,7 +119,20 @@ try:
             test_params.setdefault('timeout', "10")
             test_params.setdefault('type', 'parsing')
 
-            if len(set(req_test_params) - set(test_params.keys())) != 0:
+            # it is an error if there is a contradictory should_fail: 0 \
+            # tag and also a should_fail_because: <msg> tag
+            if ("should_fail", "0") in test_params.keys(
+            ) and "should_fail_because" in test_params.keys():
```

Comment:
Then this check could be moved to just after the `for/else` loop and I think that it could be modified to override `should_fail` based on `should_fail_because` value, ie. if the string is not empty and `should_fail` is `0` then set it to `1` and issue a warning but if string is empty and `should_fail` is `1` then only issue a warning that `should_fail` should be replaced with `should_fail_because`

Reply:
Yeah this was my initial idea, but i thought that if i added another required tag and it wasn't there the check that breaks from the loop when all tags are found would keep iterating and finally there will be an exception of missing parameter, 
Just after your comment i noticed the default value given to other tags incase they are missing,
you just made my life easier 😁